src/psyclone/psyir/backend/base.py

```diff
@@ -99,6 +102,52 @@ def __init__(self, skip_nodes=False, indent_string="  ",
         self._indent = indent_string
         self._depth = initial_indent_depth
 
+    @abc.abstractmethod
+    def codeblock_node(self, node):
```

Comment:
I've chatted with @sergisiso and @rupertford about this and now understand things a bit better. By default the visitor will raise an Exception if a handler is not found. Therefore (contrary to what I said last time) we don't need these abstract methods as they aren't going to do anything for us.

Reply:
I'm happy with this. I wasn't looking forward to add all those abstract methods 😁 So I've removed those (but left the duplicated code from c/Fortran moved into base)